lms/djangoapps/instructor/views/coupons.py

```diff
@@ -142,11 +156,14 @@ def get_coupon_info(request, course_id):  # pylint: disable=unused-argument
         return JsonResponse({
             'message': _("coupon with the coupon id ({coupon_id}) is already inactive").format(coupon_id=coupon_id)
         }, status=400)  # status code 400: Bad Request
-
+    expiry_date = 'Never Expires'
+    if coupon.expiration_date:
+        expiry_date = coupon.display_expiry_date
```

Comment:
Rather than storing this in the model, you could have the view own this logic. The "Never Expires" should be a string in the template displayed when no expiration date is found for a coupon. If there is a date, you could use something similar to this (Django Template, not 100% certain of the Mako Equivalent)
https://github.com/edx/edx-ora2/blob/master/openassessment/templates/openassessmentblock/response/oa_response.html#L21-21